signal-hook-registry/src/lib.rs

```diff
@@ -42,6 +42,15 @@
 //! require dependencies that don't build there, so tests need newer Rust version (they are run on
 //! stable).
 //!
+//! # Portability
+//!
+//! This crate includes a limited support for Windows, based on `signal`/`raise` in the CRT.
+//! There are differences in both API and behavior:
+//!
+//! - Due to lack of `siginfo_t`, we don't provide `register_sigaction` or `register_unchecked`.
+//! - Due to lack of signal blocking, there's a race condition.
+//!   After the call to `signal`, there's a moment where we miss a signal.
```

Comment:
I'd suggest listing the consequences here ‒ „If this happens, the default handler is called instead“.

Reply:
`signal()` comes before RCU store -- so I guess nothing happens (neither the default handler nor the registered handler is called) instead.

Reply:
I'm not sure I understand. Or, are we both thinking about the same race condition?

This is what I fear might happen:

* SIGTERM (or something else) arrives. This resets the signal to default, which is terminate.
* It calls the handler.
* Before the handler can re-establish itself back from the default, another SIGTERM arrives. This one terminates the program, because it didn't get re-established.

It is probably quite rare for the user to press CTRL+C *that* fast, so it most probably won't happen, but in theory something could send them programmatically?

But I'm not sure about the theory.

Reply:
Yeah I imagined a different one than yours, so I added another bullet here to describe yours. The one in my mind was:

- The program tries to register a handler for SIGINT. It acquires the RCU lock. We don't block signals here because there's no such machinery in Windows.
- `signal` is called. Our global `handler` function is registered.
- SIGINT arrives. As `handler` is already registered, the default handler isn't invoked. However, we haven't stored the RCU result yet so there's no shared handler registered.
- RCU store happens.

In this case we miss SIGINT where we expect the default handler ("before handler registration" case) or the registered handler ("after handler registration" case) to be invoked.